python/pynq/pl.py

```diff
@@ -152,6 +152,165 @@ def _get_gpio(tcl_name):
 
     return result
 
+class _InterruptMap:
+    """Helper Class to extract interrupt information from a TCL configuration file
+
+    Attributes
+    ----------
+    intc_parent : dict [str, str, int]
+        All AXI interrupt controllers in the system attached to
+        a PS7 interrupt line. Key is the name of the controller and
+        value is parent interrupt controller and the line interrupt used
+        The PS7 is the root of the hierarchy and is unnamed
+
+    intc_pins : dict [str, str, int]
+        All pins in the design attached to an interrupt controller listed in
+        in intc_names. Key is the name of the pin, value is the interrupt
+        controller and line used.
+
+    """
+
+    def __init__(self, tcl_name):
+        """Returns a map built from the supplied tcl file
+
+        Paramters
+        ---------
+        tcl_name : str
+            The tcl filename to parse. This is opened directly so should be
+            fully qualified
+
+        """
+        if not isinstance(tcl_name, str):
+            raise TypeError("tcl_name has to be a string")
+
+        # TODO: support nested hierarchies
+
+        # Initialize result variables
+        self.intc_names = []
+        self.intc_parent = {}
+        self.concat_cells = {}
+        self.nets = []
+        self.pins = {}
+        self.intc_pins = {}
+        self.ps7_name = ""
+
+        # Key strings to search for in the TCL file
+        hier_pat = "create_hier_cell"
+        concat_pat = "create_bd_cell -type ip -vlnv xilinx.com:ip:xlconcat:2.1"
+        interrupt_pat = "create_bd_cell -type ip -vlnv xilinx.com:ip:axi_intc:4.1"
+        ps7_pat = "create_bd_cell -type ip -vlnv xilinx.com:ip:processing_system7:5.5"
+        prop_pat = "set_property -dict"
+        end_prop_pat = "] $"
```

Comment:
Seems like this variable is never used? I suggest to use python editors like pycharm to easily fix this kind of problem.